ipywidgets/widgets/widget_image.py

```diff
@@ -128,3 +129,27 @@ def _guess_format(cls, filename):
             return mtype[len('image/'):]
         except Exception:
             return None
+
+    def __repr__(self):
+        # Truncate the value in the repr, since it will
+        # typically be very, very large.
+
+        def hash_string(s):
+            return hashlib.md5(s.encode("utf8")).hexdigest()
+
+        class_name = self.__class__.__name__
+
+        # Return value first like a ValueWidget
+        signature = []
+        sig_value = str(self.value)
+        if len(str(self.value)) > 100:
+            sig_value = '<base64, hash={}...>'.format(hash_string(sig_value)[:16])
```

Comment:
We don't know that it is base64, do we? It might be a very long URL.

I like your previous approach of just truncating. Or perhaps we should just enhance the existing repr keys mechanism to let us return a repr form for any key, rather than just whether to include the key or not.